consensus/types/src/light_client_header.rs

```diff
@@ -15,12 +21,160 @@ use test_random_derive::TestRandom;
     TestRandom,
     arbitrary::Arbitrary,
 )]
-pub struct LightClientHeader {
+#[serde(bound = "E: EthSpec")]
+#[arbitrary(bound = "E: EthSpec")]
+pub struct LightClientHeader<E: EthSpec> {
     pub beacon: BeaconBlockHeader,
+    #[test_random(default)]
+    #[ssz(skip_serializing, skip_deserializing)]
+    pub execution: Option<ExecutionPayloadHeader<E>>,
+    #[test_random(default)]
+    pub execution_branch: Option<FixedVector<Hash256, ExecutionPayloadProofLen>>,
 }
 
-impl From<BeaconBlockHeader> for LightClientHeader {
+impl<E: EthSpec> From<BeaconBlockHeader> for LightClientHeader<E> {
     fn from(beacon: BeaconBlockHeader) -> Self {
-        LightClientHeader { beacon }
+        LightClientHeader {
+            beacon,
+            execution: None,
+            execution_branch: None,
+        }
     }
 }
+
+impl<E: EthSpec> LightClientHeader<E> {
+    fn new(chain_spec: ChainSpec, block: SignedBeaconBlock<E>) -> Result<Self, Error> {
```

Comment:
`ChainSpec` is needed here to figure out what fork were at. Unless theres a better way to do it?

Reply:
Depends on whether we need to do a fork consistency check here - if we don't, we could use `block.fork_name_unchecked` (but I haven't read the whole thing yet)